src/app/client/src/app/modules/dial-code-search/components/dial-code/dial-code.component.ts

```diff
@@ -433,17 +433,12 @@ export class DialCodeComponent implements OnInit, OnDestroy {
   }
   public handleCloseButton() {
     if (_.get(this.activatedRoute, 'snapshot.queryParams.textbook') && _.get(this.dialCodeService, 'dialCodeResult.count') > 1) {
-      this.router.navigate(['/get/dial', _.get(this.activatedRoute, 'snapshot.params.dialCode')]);
+      return this.router.navigate(['/get/dial', _.get(this.activatedRoute, 'snapshot.params.dialCode')]);
+    }
+    if (this.userService.loggedIn) {
+      this.navigationHelperService.navigateToPreviousUrl('/resources');
```

Comment:
Why did you removed the check for "play" in the URL? Check with @yravinderkumar33 on this implementation.

Reply:
its not needed. In the history /play url will not be there. Please check the logic of navigationHelperService.